dApp/user/src/lib/Auth/AuthProvider.res

```diff
@@ -0,0 +1,41 @@
+type authContextType = {
+  isAuthorized: bool,
+  setIsAuthorized: ((bool => bool) => unit)
+}
+
+module AuthContext = {
+  let context = React.createContext({isAuthorized: false, setIsAuthorized: (_)=> () })
+
+  module Provider = {
+    let provider = React.Context.provider(context)
+
+    @react.component
+    let make = (~value, ~children) => {
+      React.createElement(provider, {"value": value, "children": children})
+    }
+  }
+}
+
+let getUserAuthStatus = (user) => Apollo.getAuthHeaders(~user)->Option.isSome
```

Comment:
Cool!

Simple and works. I was thinking, maybe we want to allow users to remain 'authenticated' even when they aren't connected via their web3 provider. This would allow a nice UX where they can still do stuff in the DB related to their user, but not be annoyed with having to open up and sign in with their web3 provider (eg metamask) first. (I'd probably say this is more needed for float, but still nice to have here)

To do this I'd create another field in local storage with the key "CurrentUserAuth". Then the concept of being 'logged in' and being 'connected to the ethereum network' become independent things. (I hope this won't add too much complexity.

Reply:
Yeah seems like a much better way of doing it - allows for a much more flexible set-up. Think we'll have to watch for syncing issues, presuming that when the user is connected with a wallet we want their wallet address to be their CurrentUserAuth, but shouldn't be too much of a hassle.

Reply:
It should sync instantly in that `useEffect` function in the provider?

Reply:
Yeah it will :sweat_smile: